lilypad/server/models/auth.py

```diff
@@ -0,0 +1,38 @@
+"""Auth models."""
+
+from collections.abc import Sequence
+from enum import Enum
+
+from pydantic import BaseModel
+from workos.organizations import Organization
+
+
+class LoginType(str, Enum):
+    """Login type enum"""
+
+    GITHUB_OAUTH = "GitHubOAuth"
+    GOOGLE_OAUTH = "GoogleOAuth"
+
+
+class UserBase(BaseModel):
+    """Base user model"""
+
+    first_name: str | None = None
+    raw_profile: dict
+    session_id: str
+    expires_at: str
+    access_token: str | None = None
+    organization_id: str | None = None
+    organizations: Sequence[Organization] | None = None
```

Comment:
could we organize all the WorkOS stuff in it's own folders (e.g. `/auth/workos` or something) and then construct the models and API and such based on an environment variable (e.g. `AUTH_PROVIDER`)?

The flip side to this question is that if we can make this change later without it being a breaking change then we can just ignore for now.

Reply:
it wont ever be a breaking change, since the user will never see this stuff. for example, in the client we would toggle. between workos and other providers, etc and we would change the url internally but that wont change the flow. get authorization url -> user authorizes -> redirect to client -> redirect to server -> sign token -> send token to user -> user uses token for requests.

Reply:
although putting this in a workos folder seems to make sense so i will do that anyways

Reply:
yeah, that makes sense. I think really I just want to do any easy work now to prep for future additional providers, but doesn't matter too much in the short term if we can introduce those changes without it breaking.

My main question is if we should have something like separate user tables partitioned by provider and then indicate which user table to use based on the environment variable or if we should have a single user table that we instantiate based on the provider. I feel like the former is more future proof?